### PR TITLE
FE-385: fix web UI query import stuck on loading for faulty JSON

### PR DESCRIPTION
**Pull request by palashkaria**
### Scope & Purpose

Fixes query import, shows JSON parse errors early

- [x] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] Manually tested
- [x] :book: CHANGELOG entry made

#### Related Information
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/FE-385
- [x] Design document:

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import { getAardvarkRouteForCurrentDb } from "../../../utils/arangoClient";
+import { useQueryContext } from "../QueryContextProvider";
+import { QueryType } from "./useFetchUserSavedQueries";
+
+const useReadFile = () => {
+  const onReadFile = ({
+    file,
+    onLoad
+  }: {
+    file?: File | null;
```
Comment (nikhil-varma) [inline js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx:12]: Optional and null?

File: js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    reader.onload = async () => {
+      const data = reader.result;
+      if (typeof data !== "string") {
+        window.arangoHelper.arangoError(`Failed to parse imported queries`);
+        return;
+      }
+      try {
+        const parsedData = JSON.parse(data);
+        onLoad(parsedData);
+      } catch (e: any) {
```
Comment (nikhil-varma) [inline js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx:29]: Not a comment but just sharing my learning 
From what I have understood (_feel free to correct me_) for errors without any type I think `unknown` is efficient to use.. `any` is just an escape hatch for TS to ignore and allow all type operations.

File: js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        if (!window.frontendConfig.ldapEnabled) {
+          setIsUploading(true);
+          await uploadToServer({
+            sanitizedQueries,
+            onSuccess: handleSuccess,
+            onFailure: handleFailure
+          });
+          return;
+        }
+        setIsUploading(true);
```
Comment (nikhil-varma) [inline js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx:101]: How about we take this outside the conditions? Line101 & 93..

**Review by nikhil-varma — COMMENTED** [review]
Minor comments/suggestions.

**Review by nikhil-varma — APPROVED** [review]

File: js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx (hunk 2 above)
Comment (palashkaria) [reply js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/useQueryImport.tsx:29]: when I do unknown, I get red swiggly lines still: 
image

**Pull request merged**